others/freivalds_algorithm.cpp

```diff
@@ -0,0 +1,177 @@
+/**
+ * @file
+ * @brief Implementation of [Freivalds' algorithm]
+ * (https://en.wikipedia.org/wiki/Freivalds%27_algorithm)
+ * @author [Darshana Sarma](https://github.com/Darshana-Sarma)
+ * @details
+ * Given three n × n matrices A, B, and
+ * C, a general problem is to verify whether AB=C
+ *
+ * The time Complexity of the algorithm is O(n^{2}).
+ */
+#include <array>     //   for std::array
+#include <cassert>   //   for assert
+#include <iostream>  //   for io operations
+
+
+/**
+ * Freivald's algorithm
+ * The function will will check AB=C
+ * @param A matrix A
+ * @param B matrix B
+ * @param C resultant matrix C
+ * @returns boolean value
+ */
+/**
+ * @namespace others
+ * @brief Other algorithms
+ */
+namespace others {
+/**
+ * @namespace freivalds
+ * @brief Functions for [Freivalds']
+ * (https://en.wikipedia.org/wiki/Freivalds%27_algorithm) algorithm
+ * @details Error Analysis: If {\displaystyle A\times B=C}{\displaystyle A\times B=C},
+ * then the algorithm always returns "Yes". If {\displaystyle A\times B\neq C}A
+ * \times B \neq C, then the probability that the algorithm returns "Yes" is less
+ * than or equal to one half. This is called one-sided error. By iterating the
+ * algorithm k times and returning "Yes" only if all iterations yield "Yes", a
+ * runtime of O(kn^{2})and error probability of <= 1/(2^k) is achieve
+ */
+namespace freivalds {
+template <std::size_t N>
+
+bool CoppersmithAlgo(std::array<std::array<double, N>, N> A,

```

Comment:
Missing function documentation: parameters, brief description and template parameters.